lib/ansible/modules/packaging/language/pip.py

```diff
@@ -519,13 +654,13 @@ def main():
                 extra_args = ' '.join(args_list)
 
         if extra_args:
-            cmd += ' %s' % extra_args
+            cmd.append(extra_args)
 
         if name:
-            for pkg in name:
-                cmd += ' %s' % _get_full_name(pkg, version)
+            for p in packages:
```

Comment:
I'd prefer
```python
# a generator expression:
cmd.extend(str(p) for p in packages)
# or:
cmd.extend(map(str, packages))
```

Reply:
A generator expression is fine here.  Just remember to use to_native instead of str().

```
cmd.extend(to_native(p, errors='surrogate_or_strict') for p in packages)
```

Reply:
A generator expression is fine here.  Just remember to use to_native instead of str().

```
cmd.extend(to_native(p, errors='surrogate_or_strict') for p in packages)
```